src/server/transaction.cc

```diff
@@ -178,13 +177,16 @@ void Transaction::InitMultiData(KeyIndex key_index) {
   // With EVAL, we call this function for EVAL itself as well as for each command
   // for eval. currently, we lock everything only during the eval call.
   if (!multi_->locks_recorded) {
+    tmp_uniques.clear();
+
     for (size_t i = key_index.start; i < key_index.end; i += key_index.step)
       lock_key(ArgS(full_args_, i));
     if (key_index.bonus)
       lock_key(ArgS(full_args_, *key_index.bonus));
+
+    multi_->locks_recorded = true;
   }
 
-  multi_->locks_recorded = true;
```

Comment:
🤔 I think this is a leftover from the incremental locking feature, we currently always acquire all locks on scheduling